### ISSUE TITLE
Black screen on Mayavi Scene 

### ISSUE THREAD

**Issue by Pouyeyazdi** [spyder-ide/spyder#13901]
**Please help me!**
I use Spyder 3.3.1 with IPython 7.16.1
I have always been using matplotlib for plotting. I decided to install mayavi. The installation was ok but when I call the library it says:

`Changing backend to Qt4 for Mayavi`

**And it opens a black screen interface!** 
When I choose QT4 for the backend of python console (in graphics) the massage does not appear but still, I have black interface! When I click for saving the snapshot, **it saves correctly the figure I did plot !! BUT** the problem is that the **interactive Mayavi scene appears broken and I can not see anything**!

----------------------------------------------------------
I checked #6339 and see that the problem should be solved for Spyder3.3 but I still get such a broken black interface!
I checked that the "**spydercustomize.py**" is here:

- ~/anaconda/pkgs/spyder-kernels-0.2.4-py36_0/lib/python3.6/site-packages/spyder_kernels/customize/spydercustomize.py

and is at it was commented in #6919
and I also checked that the "**shell.py**" is here:

- ~/anaconda/pkgs/spyder-3.3.1-py36_1/lib/python3.6/site-packages/spyder/widgets/ipythonconsole/shell.py-

and they are as it was commented in #6919

----------------------------------------------------------
Then without restarting the Kernel and without putting backend on Automatic, if I run the following lines:
```
import matplotlib.pyplot as plt
fig=plt.figure()
```

I get the error:
`ImportError: Cannot load backend 'Qt4Agg' which requires the 'qt4' interactive framework, as 'qt5' is currently running`

**I do not understand!! When the backend is QT4 it says qt5 is working!!**

----------------------------------------------------------

Another issue is that :
in my Ipython 
`import PyQt5`
is Ok, but when I do 
`import PyQt4 `
the following massage comes up:
```
Traceback (most recent call last):

  File "<ipython-input-10-98778b40e1be>", line 1, in <module>
    import PyQt4

  File "/Users/pouye/anaconda/lib/python3.6/site-packages/IPython/external/qt_loaders.py", line 57, in load_module
    """ % (fullname, loaded_api()))

ImportError: 
    Importing PyQt4 disabled by IPython, which has
    already imported an Incompatible QT Binding: pyqt5 
```
----------------------------------------------------------

Spyder version 3.3.1
Python version 3.6.10
for Qt version I did: 
```
    $ qmake -v
    QMake version 3.1
    Using Qt version 5.9.7 in ~/anaconda/lib
```

for PyQt version I did: 
```
    $ conda list
    pyqt    5.9.2
```

Operating System: macOS Catalina version10.15.7

**Comment by andfoy** [comment]
Hi @Pouyeyazdi, thanks for reporting this issue, it seems that you are using an old version of Spyder, could you please update it to the latest version (4.1.5) by executing the following commands:

```
conda update anaconda
conda install spyder=4.1.5
```

Please let us know if this fixes your issue

**Comment by Pouyeyazdi** [comment]
Unfortunately not! 
I did both update and install and now what happens is the following:
`import Mavayi`
`Changing backend to Qt for Mayavi`

And when I try to run an example it says:
`Changing backend to Qt for Mayavi`
`Qt WebEngine seems to be initialized from a plugin. Please set Qt::AA_ShareOpenGLContexts using QCoreApplication::setAttribute before constructing QGuiApplication.`

and once again the Mayavi scene is black and it can only be seen if I save the snapshot!

Moreover, I have a new problem with Basemap!! 

```
from mpl_toolkits.basemap import Basemap
Traceback (most recent call last):

  File "<ipython-input-10-d9467465a3b6>", line 1, in <module>
    from mpl_toolkits.basemap import Basemap

  File "/Users/pouye/anaconda/lib/python3.6/site-packages/mpl_toolkits/basemap/__init__.py", line 26, in <module>
    from matplotlib.cbook import dedent

ImportError: cannot import name 'dedent'
```
Please help me understand what happens? and at least how can I survive the Basemap as before?

**Comment by Pouyeyazdi** [comment]
I solved the problem this way:
Before I need to say that after doing `pip install PyQt5` Spyder stopped opening so I did `pip uninstall PyQt5`
And I had mayavi installed by pip: `pip install Mayavi`! So it was not functioning well and the scene was black.
I also add that doing `conda install -c anaconda mayavi ` as it is in the Anaconda cloud, never worked for me and it was always giving: 
```
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: - failed with repodata from current_repodata.json, will retry with next repodata source.
```
So what I did and everything suddenly start working is: 

```
pip uninstall mayavi
pip uninstall PyQt5
pip uninstall pyqtwebengine
pip install PyQt5==5.12.1 pyqtwebengine==5.12.1
pip install mayavi
```
Now spyder (version 4.1.5) opens and mayavi works and the screen is not black anymore and it shows the graph.
I also updated both matplotlib and basemap for the other issue and all are working so far.
